src/librbd/internal.cc

```diff
@@ -2872,6 +2872,55 @@ int validate_mirroring_enabled(ImageCtx *ictx) {
             << cpp_strerror(r) << dendl;
         }
       }
+    } else if (next_mirror_mode == cls::rbd::MIRROR_MODE_DISABLED) {
+
+      bool needs_rollback = false;
+      if (current_mirror_mode == cls::rbd::MIRROR_MODE_POOL) {
+        r = cls_client::mirror_mode_set(&io_ctx, cls::rbd::MIRROR_MODE_IMAGE);
+        if (r < 0) {
+          lderr(cct) << "Failed to set mirror mode to image: "
+            << cpp_strerror(r) << dendl;
+          return r;
+        }
+        needs_rollback = true;
+      }
+
+      std::vector<std::string> image_ids;
+      r = cls_client::mirror_image_list(&io_ctx, &image_ids);
```

Comment:
I'm not convinced that we should that. 
Also, why do we need to have two mirroring modes (pool / image) at the pool level? In my understanding, the mirroring state of an image is what decides if the image is mirrored or not, so why not just have enabled/disable states for the pool as well?